cantools/database/can/c_source.py

```diff
@@ -843,39 +842,26 @@ def _format_comment(comment):
         return ''
 
 
-def _format_decimal(value, is_float=False, use_float=False):
-    f_append = 'f' if use_float else ''
-    if int(value) == value:
-        value = int(value)
-
-        if is_float:
-            return f'{value}.0{f_append}'
-        else:
-            return str(value)
-    else:
-        return f'{value}{f_append}'
-
-
 def _format_range(signal):
     minimum = signal.minimum
     maximum = signal.maximum
 
+    phys_to_raw = float if signal.is_float else round
+
     if minimum is not None and maximum is not None:
-        return '{}..{} ({}..{} {})'.format(
-            _format_decimal(signal._signal.scaled_to_raw(minimum)),
-            _format_decimal(signal._signal.scaled_to_raw(maximum)),
-            minimum,
-            maximum,
-            signal.unit)
+        return \
+            f'{phys_to_raw(signal._signal.scaled_to_raw(minimum))}..' \
```

Comment:
What's the purpose of this? `phys_to_raw(signal._signal.scaled_to_raw(minimum))`

Reply:
this ensures that the minimum/maximum printed here are of the correct type: for some signals, it can happen that the minimum or maximum is specified as an integer but the signal is defined to internally be `float` (i.e., `signal.is_float` is true, yet `isinstance(signal.maximum, int)` also applies. this happens for example with the `S9` signal in [`tests/files/dbc/multiplex_2.dbc`](https://github.com/cantools/cantools/blob/master/tests/files/dbc/multiplex_2.dbc).) While I consider this a minor bug in the file format loaders where this happens, I was not in the mood of hunting it down and the this work-around is not too bad IMO, mainly because it makes the code a bit more fool-proof...

In case you are wondering about `signal._signal`: the `signal` object here is of type [`Signal` from `c_source.py`](https://github.com/cantools/cantools/blob/master/cantools/database/can/c_source.py#L582), and the conversion stuff did not work for it for some reason which I forgot. (I think this code should be refactored...)

Reply:
ok, I renamed `phys_to_raw` to `cast_to_raw_type`. I hope the new name is much clearer...

Reply:
I also fixed the minor bug in the conversion code which caused this: https://github.com/cantools/cantools/pull/555/commits/084d48e83a47a7935659b36049935fb18b1b55cb